docs/zkapps/o1js/common-types-and-functions.md

```diff
@@ -0,0 +1,354 @@
+  - common types
+  - built in types
+  - bool
+  - uint32
+  - uint64
+  - provable if
+  - loops
+  - functions
+---
+
+:::info
+
+zkApp programmability is not yet available on the Mina Mainnet. You can get started now by deploying zkApps to the Berkeley Testnet.
+
+:::
+
+# o1js Built-in Data Types
+
+o1js includes built in types and methods that you can use inside your zkApp and ZKPs (zero knowledge proofs). All o1js provable types are built using the [Field](./basic-concepts.md#field) type, each with unique properties and advantages. You can find detailed class description in [o1js reference](../o1js-reference).
+
+## Bool
+
+`Bool` is the type used for `boolean` values. It is equivalent to the `boolean` type in order languages and frameworks.
+
+<!-- prettier-ignore -->
+```ts
+const x = new Bool(true);
+const y = new Bool(false);
+```
+
+With the `Bool` type, you can call the contructor without `new`, just like the `Field` type. The best practice is to use the constructor function without `new`:
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(false);
+```
+
+You can also create a `Bool` from another `Bool`.
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(x);
+```
+
+It is also possible to convert a `Field` into a `Bool`. However, you should make sure that the `Field` is "bool-like" (with a value either 1 or 0) before converting it.
+
+```ts
+let x = Field(1);
+x.assertBool(); // Asserts that x has a value either 1 or 0
+
+let x_bool = Bool(x.value);
+```
+
+`Bool` type includes the following conditional methods and converters. Every method created and returns a new `Bool`. All conditional methods can be called with regular `boolean` values as well as `Bool` variables:
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(false);
+const z = y.not(); // True
+
+const equals = x.equals(y); // False
+const and = x.and(y); // False
+const or = x.or(y); // True
+
+const equals_from_boolean = x.equals(true); // True
+const and_from_boolean = x.and(true); // True
+const or_from_boolean = y.or(false); // False
+```
+
+You can also assert `Bool` variables. When you assert something, your zkApp method throws if the assertion fails.
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(false);
+
+x.assertEquals(y); // Throws
+x.assertTrue(); // Nothing happens, as assertion passes
+x.assertFalse(); // Throws
+```
+
+You can also combine multiple logical expressions one after another.
+
+<!-- prettier-ignore -->
+```ts
+const x = Bool(true);
+const y = Bool(false);
+
+x.not().and(y).or(true); // True
+```
+
+**Important:** All logical expressions are executed in the method call order, as with any other regular language or framework. However, as these are logical statements inside ZKPs, all expressions are always evaluated regardless of the logical ordering. Please see conditionals chapter [below](./common-types-and-functions.md#conditionals) for more details.
```

Comment:
```suggestion
**Important:** All logical expressions are executed in the method call order, as with any other regular language or framework. However, as these are logical statements inside ZKPs, all expressions are always evaluated regardless of the logical ordering. See [Conditionals](./common-types-and-functions.md#conditionals) for more details.
```

Reply:
I agree with Gregor, nice work @yunus433 !
Guidance is to omit "please", see [Politeness and use of please](https://developers.google.com/style/tone#politeness-and-use-of-please)

Instead of "below" or "here" be sure to use short [descriptive link text](https://developers.google.com/style/link-text) that helps provide context for the material that you're linking to.